### PR TITLE
Tidy up: centralise parser loader so this can be called by data_type

### PR DESCRIPTION
**Pull request by rt17603**
* **Summary of changes** (Bug fix, feature, docs update, ...)

Similarly to #1094 , this adds two new functions which allow the parse function itself to be loaded by supplying the `data_type` and `source_format` and uses these within the relevant store classes. This is, again, so that only the data_type input is needed to ensure the correct details can be accessed within the read_file and transform_data methods. The specific functions e.g. `load_surface_parser` have been removed.

The two functions are:
 - `openghg.util.load_standardise_parser`
 - `openghg.util.load_transform_parser`

This impacts the same data_type objects as PR #1094 .

For the `Footprints` object this had to have some additional small changes as the standardise sub-module was called `footprint` (singular) but the data_type was called `footprints` meaning these two did not match. Change made:
 - Renamed standardise sub-module to `footprints` rather than `footprint` (likewise with tests)

*Note: This PR does not relocate these parsers loaders from util but this could potentially be improved by moving these functions to `openghg.standardise` if this doesn't cause circular imports when accessing this from `openghg.store`*

* **Please check if the PR fulfills these requirements**

- [x] Relevant to #1071 but does not close this
- [x] [Tests added and passed](https://docs.openghg.org/development/python_devel.html#testing) if fixing a bug or adding a new feature
- [x] All code checks passing - `black --line-length 110` run over code, `mypy` and `flake8` not showing any errors

**Do we want this?**
- [ ] Added an entry in the latest `CHANGELOG.md` file if fixing a bug or adding a new feature

Internal - not needed
- ~Documentation and tutorials updated/added~
- ~Added any new requirements to `requirements.txt` and `recipes/meta.yaml`~

### CODE REVIEW

**Review by brendan-m-murphy — CHANGES_REQUESTED** [review]
Looks good, although I think the parser loading functions shouldn't be in `util` since they're only used by `store`. It might make sense to have them in `store` or in `standardise` with the parsers, since they provide access to the parsers (this would be my preference).

File: openghg/util/_file.py [hunk 1]
```diff
@@ -7,11 +7,8 @@
 
 __all__ = [
     "load_parser",
```
[the remark this answers is not in the document]
Comment (brendan-m-murphy) [reply openghg/util/_file.py:9]: Ah sorry I didn't see that comment. No it doesn't have to happen in this PR. I'll approve.

**Review by brendan-m-murphy — DISMISSED** [review]

File: openghg/util/_file.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (rt17603) [reply openghg/util/_file.py:9]: That's fine - it's a good suggestion and I think we will need to do it at some point.

**Review by brendan-m-murphy — APPROVED** [review]
Reapproving after merging in devel

**Pull request merged**